src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -34,7 +34,31 @@ public async Task RunAsync(ActionRunStage stage)
             Trace.Entering();
             ArgUtil.NotNull(ExecutionContext, nameof(ExecutionContext));
             ArgUtil.NotNull(Inputs, nameof(Inputs));
-            ArgUtil.NotNull(Data.Steps, nameof(Data.Steps));
+
+            List<Pipelines.ActionStep> steps;
+            if (stage == ActionRunStage.Pre)
+            {
+                ArgUtil.NotNull(Data.PreSteps, nameof(Data.PreSteps));
+                steps = Data.PreSteps;
+            } 
+            else if (stage == ActionRunStage.Post)
+            {
+                ArgUtil.NotNull(Data.PostSteps, nameof(Data.PostSteps));
+                steps = Data.PostSteps.ToList();
```

Comment:
nit: as a reader stacks and queues never cease to confuse me what their order is when `.ToList()`. Wonder whether using a list instead would simplify this